stdgo/reflect/Reflect.hx

```diff
@@ -654,3 +841,144 @@ function nameToGo(n:GoString):GoString {
+	var val:Any;
+	var surfaceType:Type;
+	var underlyingType:Type;
+
+	public function new(?v:Any = null, t:Type = GT_invalid) {
+		val = v;
+		surfaceType = t;
+		underlyingType = findUnderlying(t);
+	}
+
+	static function findUnderlying(t:Type):Type{
+		switch (t) {
+			case GT_namedType(name):
+				return findUnderlying(_Type__extension.getNamedTypeInfo(name).underlying);
+			case _:
+				return t;
+		}
+	}
+
+	@:to function __promote()
+		return new AnyInterface({value: this, typeName: Type.getClassName(Value)});
+
+	public inline function type()
+		return surfaceType;
+
+	public inline function kind()
+		return surfaceType.kind();
+
+	public inline function interface_()
+		return new AnyInterface({value: val, typeName: surfaceType.serialize()});
+
+	public inline function isNil()
+		return val == null;
+
+	public inline function isValid()
+		return surfaceType != GT_invalid;
+
+	public function bool():Bool {
+		switch (underlyingType) {
+			case GT_bool:
+				return (val : Bool);
+			case _:
+				throw "Value.Bool not implemented for " + underlyingType.string();
+		}
+		return false;
+	}
+
+	public function int():GoInt64 {
+		switch (underlyingType) {
+			case GT_int, GT_int8, GT_int16, GT_int32, GT_int64:
+				return (val : GoInt64);
+			case _:
+				throw "Value.Int not implemented for " + underlyingType.string();
+		}
+		return 0;
+	}
+
+	public function uint():GoUInt64 {
+		switch (underlyingType) {
+			case GT_uint, GT_uint8, GT_uint16, GT_uint32, GT_uint64:
+				return (val : GoUInt64);
+			case _:
+				throw "Value.Uint not implemented for " + underlyingType.string();
+		}
+		return 0;
+	}
+
+	public function float():GoFloat64 {
+		switch (underlyingType) {
+			case GT_float32, GT_float64:
+				return (val : GoFloat64);
+			case _:
+				throw "Value.Float not implemented for " + underlyingType.string();
+		}
+		return 0;
+	}
+
+	public function complex():GoComplex128 {
+		switch (underlyingType) {
+			case GT_complex128, GT_complex64:
+				return (val : GoComplex128);
+			case _:
+				throw "Value.Complex not implemented for " + underlyingType.string();
+		}
+		return 0;
+	}
+
+	public function string():GoString {
+		switch (underlyingType) {
+			case GT_string:
+				return (val : GoString);
+			case _:
+				throw "Value.String not implemented for " + underlyingType.string();
+		}
+		return "";
+	}
+}
+
+//---------------------------------
+// reflect.HaxeValue implementation
+// only method is String()
+//---------------------------------
+var haxeTypeUnknown = GT_namedType("stdgo.reflect._HaxeValue__extension");
+
+@:using(stdgo.reflect.Reflect._HaxeValue__extension)
+abstract HaxeValue(Any) from(Any) to(Any) {
+	public inline function new(?a:Any = null) {
+		this = a;
+	}
+
+	@:to inline function __promote() // this is why Kind is an abstract
+		return new AnyInterface({value: this, typeName: Type.getClassName(_HaxeValue__extension)});
+}
+
+@:rtti
+class _HaxeValue__extension {
+	static final _typeName_ = "unsafePointer";
```

Comment:
Shouldn't the typeName be "unsafe.Pointer"?

Reply:
`unsafe.Pointer` has its own Kind, and so its own GT_enum value: GT_unsafePointer. The typeName in this example is just the text of that value after "GT_".